pkg/sanity/resources.go

```diff
@@ -299,31 +296,33 @@ func (cl *Resources) cleanupVolume(ctx context.Context, offset int, volumeID str
 				Secrets:  cl.Context.Secrets.ControllerUnpublishVolumeSecret,
 			},
 		)
-		logger.Errorf(err, "ControllerUnpublishVolume failed: %s", err)
+		ExpectWithOffset(offset, err).NotTo(HaveOccurred(), "[volume cleanup] ControllerUnpublishVolume failed")
```

Comment:
Previously errors during cleanup were logged, but cleanup continued, right? Now it aborts.

Error handling during cleanup is tricky. Ideally, as much as possible should get cleaned up and then a summary of all unsolved problems should be raised as a test failure. The latter ensures that a test that itself ran fine but then has problems during cleanup is reported as failed overall.

Reply:
Whoops, somehow I missed that we used to continue to clean up even when we hit errors.

I made another update to collect all errors and assert at the end of the `Cleanup` call. I think this goes even beyond what we had before, where we'd assert at the end of a `cleanup{Volume,Snapshot}` call.

One thing I want to mention is that failure to, say, node-unpublish a volume, could mean that follow-up cleanup operations on the same resource (node-unstage, controller-unpublish, delete) may then be more likely to fail as well. I'm wondering if this aggravates understanding the produced error messages, and whether it could make more sense to not try follow-up cleanup operations within a single resource (but continue to try to clean up all resources).

I followed your suggestion for now, but let me know if you have thoughts on the above and whether it should influence another update.

Reply:
I agree that errors are likely to cause other errors. But that is nothing new, developers should be familiar with follow-up errors and focused on the first one when trying to understand what went wrong.

So I think this behavior is fine.